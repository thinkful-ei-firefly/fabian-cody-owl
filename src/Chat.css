.chat{
  /*display: none;
   display: flex; */
  flex-direction: column;
}

.chat-message{
  display: flex;
  border-color: aqua;
  flex-direction: row;
}

.chat-message img {
  border-radius: 50%; 
  width:40px; 
  height:40px; 
}

.chat-message p, h4{
  margin: 2px;
}

.system-message{
  display: flex;
  color: rgb(88, 88, 88);
  flex-direction: row;
  justify-content: space-between;
}

.system-message *{
  border: none;
}