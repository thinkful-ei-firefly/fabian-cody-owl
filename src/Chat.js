import React from 'react';
import './Chat.css';

class Chat extends React.Component {

	render() {
		const date = new Date(this.props.timestamp);
		const time = date.toLocaleTimeString('en-US', { hour: 'numeric', hour12: true, minute: 'numeric' });

		if (this.props.type === 'message') {
			return (
				<div className='chat-message'>
					<img src={this.props.participant.avatar} alt='avatar' />
					<div>
						<h4>{this.props.participant.name} <span>{time}</span></h4>
						<p>{this.props.message}</p>
					</div>
				</div>
			);
		} else {
			return (
				<div className='system-message'>
					<span>{this.props.participant.name}</span>
					<div>'{this.props.type}'</div>
				</div>
			);
		}

	}
}

export default Chat;