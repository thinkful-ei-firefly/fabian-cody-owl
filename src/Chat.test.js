import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Chat from './Chat';

describe('<Chat.js />', () => {
  it('<Chat.js /> renders properly', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Chat
      message="hello msg"
      participant={{
        id: 3,
        name: 'Axe Kubicka',
        avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
        inSession: false,
        onStage: false
      }}
      type='thumbs-up'
      time={new Date(1548852484247).toLocaleTimeString('en-US', { hour: 'numeric', hour12: true, minute: 'numeric' })}
      timestamp={1548852484247}
    />, div)
  })

  it('Renders <Chat.js /> as expected', () => {
    const tree = renderer.create(<Chat 
      message="hello msg"
      participant={{
        id: 3,
        name: 'Axe Kubicka',
        avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
        inSession: false,
        onStage: false
      }}
      type='thumbs-up'
      time={new Date(1548852484247).toLocaleTimeString('en-US', { hour: 'numeric', hour12: true, minute: 'numeric' })}
      timestamp={1548852484247}
    />).toJSON();

    expect(tree).toMatchSnapshot();
  })
})