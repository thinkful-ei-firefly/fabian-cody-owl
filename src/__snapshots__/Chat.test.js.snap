// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat.js /> Renders <Chat.js /> as expected 1`] = `
<div
  className="system-message"
>
  <span>
    Axe Kubicka
  </span>
  <div>
    '
    thumbs-up
    '
  </div>
</div>
`;
